volume_provider/providers/aws.py

```diff
@@ -143,9 +144,13 @@ def _resize(self, volume, new_size_kb):
         new_size_gb = volume.convert_kb_to_gb(new_size_kb)
         self.client.ex_modify_volume(ebs, {'Size': new_size_gb})
 
-    def _take_snapshot(self, volume, snapshot):
+    def _take_snapshot(self, volume, snapshot, team, engine, db_name):
         ebs = self.__get_ebs(volume)
-        new_snapshot = self.client.create_volume_snapshot(ebs)
+        ex_metadata = TeamClient.make_tags(team, engine)
+        ex_metadata.update({'Bkp_DBaaS': 1,'engine': engine,'db_name': db_name,
```

Comment:
Check if team, engine or db_name exists to create the dict key. If u dont verity that, the tags will be created with None